packages/hawtio/src/preferences/HomePreferences.tsx

```diff
@@ -1,13 +1,46 @@
 import { Button, CardBody, Form, FormGroup, FormSection, Switch } from '@patternfly/react-core'
 import React, { useState } from 'react'
-import { log } from './globals'
+
+type LocalStorageFieldData<T> = {
+  localStorageKey: string
+  defaultValue: T
+  updateFunction: React.Dispatch<React.SetStateAction<T>>
+}
 
 export const HomePreferences: React.FunctionComponent = () => {
-  const [defaultVerticalNavState, setDefaultVerticalNavState] = useState(true)
+  const LOCAL_STORAGE_VERTICAL_NAVIGATION = 'local_storage_vertical_navigation'
+  const DEFAULT_VALUE_VERTICAL_NAVIGATION = true
+
+  const VERTICAL_NAVIGATION_INITIAL_VALUE: boolean =
+    localStorage.getItem(LOCAL_STORAGE_VERTICAL_NAVIGATION) !== null
+      ? localStorage.getItem(LOCAL_STORAGE_VERTICAL_NAVIGATION) === 'true'
+      : DEFAULT_VALUE_VERTICAL_NAVIGATION
+
+  const [defaultVerticalNavState, setDefaultVerticalNavState] = useState(VERTICAL_NAVIGATION_INITIAL_VALUE)
+
+  const VERTICAL_NAVIGATION_FIELD_PARAMETERS: LocalStorageFieldData<boolean> = {
```

Comment:
I'd like to have this at the top, instead of having to set all those consts first before hand, make this kind of field the "const" for each variable to be saved in local storage.

But I didn't find any good way, as I need to compute the initialization value to retrieve the update function, and then I can create this object.

If any one has any good ideas please chime in

Reply:
I don't see why we need this. It seems it just makes a simple thing a bit more complicated. You can simply persist the vertical nav state to local storage via the preferences service.